src/Microsoft.ML.Transforms/GcnTransform.cs

```diff
@@ -225,477 +255,725 @@ public void Save(ModelSaveContext ctx)
+            env.CheckValue(args, nameof(args));
+            env.CheckValue(input, nameof(input));
 
-            // REVIEW: for now check only global (default) values. Move to Bindings/ColInfoEx?
-            if (!args.SubMean && args.UseStdDev)
+            env.CheckValue(args.Column, nameof(args.Column));
+            var cols = new GcnColumnInfo[args.Column.Length];
+            using (var ch = env.Start("ValidateArgs"))
             {
-                using (var ch = Host.Start("Argument validation"))
+                for (int i = 0; i < cols.Length; i++)
                 {
-                    ch.Warning("subMean parameter is false while useStd is true. It is advisable to set subMean to true in case useStd is set to true.");
+                    var item = args.Column[i];
+                    cols[i] = new GcnColumnInfo(item.Source ?? item.Name,
+                        item.Name,
+                        item.SubMean ?? args.SubMean,
+                        item.UseStdDev ?? args.UseStdDev,
+                        item.Scale ?? args.Scale);
                 }
+                if (!args.SubMean && args.UseStdDev)
+                    ch.Warning("subMean parameter is false while useStd is true. It is advisable to set subMean to true in case useStd is set to true.");
             }
-            SetMetadata();
+            return new LpNormalizingTransform(env, cols).MakeDataTransform(input);
         }
 
-        /// <summary>
-        /// A helper method to create LpNormNormalizer transform for public facing API.
-        /// </summary>
-        /// <param name="env">Host Environment.</param>
-        /// <param name="input">Input <see cref="IDataView"/>. This is the output from previous transform or loader.</param>
-        /// <param name="name">Name of the output column.</param>
-        /// <param name="source">Name of the column to be transformed. If this is null '<paramref name="name"/>' will be used.</param>
-        /// <param name="normKind">The norm to use to normalize each sample.</param>
-        /// <param name="subMean">Subtract mean from each value before normalizing.</param>
-        public static IDataTransform CreateLpNormNormalizer(IHostEnvironment env,
-            IDataView input,
-            string name,
-            string source = null,
-            NormalizerKind normKind = Defaults.NormKind,
-            bool subMean = Defaults.LpSubMean)
-        {
-            var args = new Arguments()
-            {
-                Column = new[] { new Column(){
-                        Source = source ?? name,
-                        Name = name
-                    }
-                },
-                SubMean = subMean,
-                NormKind = normKind
-            };
-            return new LpNormNormalizerTransform(env, args, input);
+        // Factory method for SignatureDataTransform for Arguments class/>
+        public static IDataTransform Create(IHostEnvironment env, Arguments args, IDataView input)
```

Comment:
>public [](start = 8, length = 6)

private!

Reply:
Why can't it be made private?

---
In reply to: [232437109](https://github.com/dotnet/machinelearning/pull/1582#discussion_r232437109) [](ancestors = 232437109)

Reply:
Because this constructor used in entrypoints.

---
In reply to: [233616058](https://github.com/dotnet/machinelearning/pull/1582#discussion_r233616058) [](ancestors = 233616058,232437109)